field.h

```diff
@@ -311,8 +311,8 @@ struct processor {
 	uint32_t set_group_used_zones{ 0 };
 	uint8_t set_group_seq[7]{};
 	uint8_t dice_result[5]{};
-	uint8_t coin_result[MAX_COIN_COUNT]{};
-	int32_t coin_count{ 0 };
+	uint32_t coin_result{ 0 };
+	uint8_t coin_count{ 0 };
```

Comment:
```suggestion
	int32_t coin_count{ 0 };
```

I suggest using int32_t for counting variable whenever possible.